src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -138,15 +141,12 @@ class EntityClient (requestContext: RequestContext)(implicit protected val execu
    * Bails with a 400 Bad Request if the entity type is unknown or if some attributes are missing.
    * Returns the list of required attributes if all is well. */
   private def withRequiredAttributes(entityType: String, headers: Seq[String])(op: (Map[String, String] => Future[PerRequestMessage])):Future[PerRequestMessage] = {
-    ModelSchema.getRequiredAttributes(entityType) match {
-      case Failure(regret) => Future(RequestCompleteWithErrorReport(BadRequest, regret.getMessage))
-      case Success(requiredAttributes) =>
-        if( !requiredAttributes.keySet.subsetOf(headers.toSet) ) {
-          Future( RequestCompleteWithErrorReport(BadRequest,
-            "TSV is missing required attributes: " + (requiredAttributes.keySet -- headers).mkString(", ")) )
-        } else {
-          op(requiredAttributes)
-        }
+    val requiredAttributes = modelSchema.getRequiredAttributes(entityType)
```

Comment:
Since `getRequiredAttributes` now throws an exception (from `getTypeSchema`) instead of returning `Failure`, I think this will result in 500 (internal server error) instead of 400 (bad request) when the entity type doesn't exist in the FC schema.